### PR TITLE
feat: add enable option for plugins

### PR DESCRIPTION
**Pull request by saihaj**
Add `enable` option to plugin which can be used to conditionally enable plugins.

### CODE REVIEW

**Comment by PabloSzx** [comment]
isn't it better to allow "null | undefined | boolean" in the list of plugins and when envelop reads the list it filters them? that's a very common usage pattern for plugins

for example, this is what rollup does, (and graphql-ez too)
image


this allows for:

```ts
[process.env.ENABLE_TRACING ? useTracing() : false, useSchema(schema)]
```

or 
```ts
[process.env.ENABLE_TRACING && useTracing(), useSchema(schema)]
```

or equivalents

**Comment by saihaj** [comment]
>  ```ts
> [process.env.ENABLE_TRACING && useTracing(), useSchema(schema)]
> ```

This looks clean we can do that too.

**Comment by dotansimha** [comment]
Thanks @saihaj ! 
I was thinking that maybe we can export a helper like: 

```
import { enableIf } from '@envelop/core';

const getEnveloped = envelop({
  plugins: [
    enableIf(isDev, somePlugin)
  ]
});
```

Then, the core can just filter the `null` plugins. This way this mechanism will support all plugins, without the need to modify all of them.

What do you think?

**Comment by saihaj** [comment]
yea that is better I will update this

File: packages/types/src/utils.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ export type TuplifyUnion<T, L = LastOf<T>, N = [T] extends [never] ? true : fals
   : Push<TuplifyUnion<Exclude<T, L>>, L>;
 
 export type Unarray<T> = T extends Array<infer U> ? U : T;
+export type NullableArray<T> = T extends Array<infer U> ? (U | null | undefined)[] : T[];
```
Comment (dotansimha) [inline packages/types/src/utils.ts]: Doing `NullableArray<PluginType>` will result in `PluginType[]` without `null`, is that intended?

Comment (saihaj) [reply packages/types/src/utils.ts]: Yes this is just to `null` values in an array so if it just a primitive it shouldn’t do anything. So it should just return `T`. I think better is `Nullable<T>` will fix it

Comment (saihaj) [reply packages/types/src/utils.ts]: What do you think if you don’t pass  an array should it return it back as is Ike null the values?

Comment (dotansimha) [reply packages/types/src/utils.ts]: It can just be: `type NullableArray<T> = Array<T | null>`?

**Comment by dotansimha** [comment]
@saihaj can you please review my changes?
I created a branded nomial value instead of `null` because it felt a bit weird to allow external API to use `null` as value. I used a private nominal value and added a helper function based on TS type guard. 

Also, pushed a few minor fixes to fix circular imports and issues with missing peerDependencies.

**Comment by saihaj** [comment]
Lgtm. That’s very smart 😎

**Pull request merged**